run_reg_test.py

```diff
@@ -0,0 +1,362 @@
+        elif args.check_golden:
+            tests_run = True
+            num_qor_failures = parse_single_test(
+                collect_task_list(reg_test), check=True
+            )
+        elif args.create_golden:
+            # Create golden results
+            num_qor_failures = 0
+            parse_single_test(collect_task_list(reg_test), create=True)
+        elif args.calc_geomean:
+            # Calculate geo mean values
+            num_qor_failures = 0
+            parse_single_test(collect_task_list(reg_test), calculate=True)
+        elif args.display_qor:
+            num_qor_failures = display_qor(reg_test)
+        elif reg_test.startswith("odin"):
+            total_num_func_failures += run_odin_test(args, reg_test)
+        else:
+            # Run any ODIN tests
+            print("=============================================")
+            print("    Verilog-to-Routing Regression Testing")
+            print("=============================================")
+
+            tests_run = True
+
+
+            # Collect the task lists
+            vtr_task_list_files = collect_task_list(reg_test)
+
+            # Run the actual tasks, recording functionality failures
+            if vtr_task_list_files:
+                num_func_failures += run_tasks(args, vtr_task_list_files)
+
+            # Check against golden results
+            if not args.skip_qor and vtr_task_list_files:
+                num_qor_failures += parse_single_test(
+                    vtr_task_list_files, check=True, calculate=True
+                )
+                print(
+                    "\nTest '{}' had {} qor test failures".format(
+                        reg_test, num_qor_failures
+                    )
+                )
+            print(
+                "\nTest '{}' had {} run failures\n".format(reg_test, num_func_failures)
+            )
+        total_num_func_failures += num_func_failures
+        total_num_qor_failures += num_qor_failures
+    # Final summary
+    if (
+        tests_run
+        and total_num_func_failures == 0
+        and (total_num_qor_failures == 0 or args.skip_qor)
+    ):
+        print("All tests passed")
+    elif tests_run and total_num_func_failures != 0 or total_num_qor_failures != 0:
+        print(
+            "Error: {} tests failed".format(
+                total_num_func_failures + total_num_qor_failures
+            )
+        )
+
+    sys.exit(total_num_func_failures + total_num_qor_failures)
+
+
+def display_qor(reg_test):
+    """ Display the qor tests script files to be run outside of this script """
+    test_dir = Path(find_vtr_root()) / "vtr_flow/tasks/regression_tests" / reg_test
+    if not (test_dir / "qor_geomean.txt").is_file():
+        print("QoR results do not exist ({}/qor_geomean.txt)".format(str(test_dir)))
+        return 1
+    print("=" * 121)
+    print("\t" * 6, end="")
+    print("{} QoR Results".format(reg_test))
+    print("=" * 121)
```

Comment:
Should the width be parameterized? Or at least add a function to print a header (lines 229-232.)

Reply:
This has been fixed